lib/matplotlib/legend.py

```diff
@@ -553,6 +562,28 @@ def __init__(self, parent, handles, labels,
         self.set_title(title, prop=tprop)
         self._draggable = None
 
+        # set the text color
+        if textcolor is None:
+            pass
+        elif textcolor == 'linecolor':
+            for line, text in zip(self.get_lines(), self.get_texts()):
```

Comment:
Will this also work for patches and other none `Line2D` entries?

Reply:
This should now work for patches with the latest commit. I think that `Line2D` and `Patch` objects are the only things that can be legend handles, right?